### PR TITLE
automatically subscribe to the file uploading events

### PR DESCRIPTION
**Pull request by KutsenkoA**

### CODE REVIEW

File: src/api/fileops/fileset.ts [hunk 1]
```diff
@@ -37,8 +46,15 @@ export class Fileset<FormDataType extends IFormData<F>, T, D, F> implements IRes
    * Upload files and create records in the fileset
    * @param files {Array<FilesetMultipart<T, F>>} list of files with record data
    */
-  public upload(files: Array<FilesetMultipart<T, F>>): Observable<T> {
-    return this.fileUploader.upload(files);
+  public upload(files: Array<FilesetMultipart<T, F>>): Observable<FilesetInterface<T>> {
+
+    let fileUploadObservable = this.fileUploader.upload(files);
+
+    if (this.clientConfig.fileOperations.subscribeForTheFileUploading) {
```
Comment (marlosin) [inline src/api/fileops/fileset.ts]: suggestion: returning without `else` looks more readable IMHO:
```js
if (this.clientConfig.fileOperations.subscribeForTheFileUploading) {
  return this.getFileUpdates(fileUploadObservable, files.length);
}

return fileUploadObservable;

File: src/api/fileops/fileset.ts [hunk 2]
```diff
@@ -47,4 +63,44 @@ export class Fileset<FormDataType extends IFormData<F>, T, D, F> implements IRes
    * public update(data: T): UpdateQuery<T> {}
    * public delete(): DeleteQuery<D> {}
    */
+
+  /**
+   * Subscribe for the RTC records
+   * update file record with a status
+   * @param uploadingProcess
```
Comment (marlosin) [inline src/api/fileops/fileset.ts]: add type for these parameters

File: e2e/stories/jfs.e2e.ts [hunk 3]
```diff
@@ -79,6 +76,9 @@ describe('Fileset Module', () => {
           default:
             finishTest('Wrong action received');
         }
+        if (messagesReceived === 2) {
```
[the remark this answers is not in the document]
Comment (KutsenkoA) [reply e2e/stories/jfs.e2e.ts]: yeap 👍

File: src/api/fileops/fileset.ts (hunk 1 above)
Comment (KutsenkoA) [reply src/api/fileops/fileset.ts]: 👍

File: src/api/fileops/fileset.ts (hunk 2 above)
Comment (KutsenkoA) [reply src/api/fileops/fileset.ts]: done

File: src/api/fileops/fileset.ts [hunk 4]
```diff
@@ -47,4 +63,44 @@ export class Fileset<FormDataType extends IFormData<F>, T, D, F> implements IRes
    * public update(data: T): UpdateQuery<T> {}
    * public delete(): DeleteQuery<D> {}
    */
+
+  /**
+   * Subscribe for the RTC records
+   * update file record with a status
+   * @param uploadingProcess {Observable<FilesetInterface<T>>} Observable of the uploading files process
```
Comment (marlosin) [inline src/api/fileops/fileset.ts]: the `type` comes comes before the param name, like:
```js
* @param {Observable<FilesetInterface<T>>} uploadingProcess Observable of the uploading files process

Comment (marlosin) [reply src/api/fileops/fileset.ts]: nevermind, the doc will be generated based on the type defined in typescript :+1:

**Pull request merged**